scripts/generateAssetData/generateRelatedAssetIndex/generateRelatedAssetIndex.ts

```diff
@@ -196,26 +211,29 @@ const processRelatedAssetIds = async (
 
   const zerionRelatedAssetIds = relatedAssetsResult?.relatedAssetIds ?? []
   const mergedRelatedAssetIds = Array.from(
-    new Set([...manualRelatedAssetIds, ...zerionRelatedAssetIds]),
+    new Set([...manualRelatedAssetIds, ...zerionRelatedAssetIds, assetId]),
   )
 
   // Has zerion-provided related assets, or manually added ones
-  const hasRelatedAssets = mergedRelatedAssetIds.length > 0
+  const hasRelatedAssets = mergedRelatedAssetIds.length > 1
 
-  // attach the relatedAssetKey for all related assets including the primary implementation (where supported by us)
-  if (hasRelatedAssets && assetData[relatedAssetKey] !== undefined) {
-    assetData[relatedAssetKey].relatedAssetKey = relatedAssetKey
-  }
-
-  for (const assetId of mergedRelatedAssetIds) {
-    assetData[assetId].relatedAssetKey = relatedAssetKey
+  if (hasRelatedAssets) {
+    // attach the relatedAssetKey for all related assets including the primary implementation
+    for (const relatedAssetId of mergedRelatedAssetIds) {
+      if (assetData[relatedAssetId]) {
+        assetData[relatedAssetId].relatedAssetKey = relatedAssetKey
+      }
+    }
+    relatedAssetIndex[relatedAssetKey] = mergedRelatedAssetIds
+  } else {
+    // If there are no related assets, set relatedAssetKey to null
+    assetData[assetId].relatedAssetKey = null
   }
-  relatedAssetIndex[relatedAssetKey] = mergedRelatedAssetIds
-  return
-}
 
-export const generateRelatedAssetIndex = async () => {
-  console.log('generateRelatedAssetIndex() starting')
+  await throttle()
```

Comment:
Note we now throttle at the end of this method - no need to throttle if we're skipping the current fetching run.